packages/runtime-common/realm.ts

```diff
@@ -239,32 +247,31 @@ export class Realm {
     }
 
     // add the node stream to the response which will get special handling in the node env
-    let response = new Response(null, {
-      headers: {
-        "last-modified": formatRFC7231(ref.lastModified),
-        vary: "Accept",
-      },
-    }) as ResponseWithNodeStream;
+    response.headers.set("last-modified", formatRFC7231(ref.lastModified));
     response.nodeStream = ref.content;
     return response;
   }
 
-  private async upsertCardSource(request: Request): Promise<Response> {
+  private async upsertCardSource(
+    request: Request,
+    response: Response
+  ): Promise<Response> {
     let { lastModified } = await this.write(
       this.paths.local(new URL(request.url)),
       await request.text()
     );
     return new Response(null, {
```

Comment:
this is an example of a response that should not be instantiate. There should be no new Response, rather the response that is used should be the one that is passed in.